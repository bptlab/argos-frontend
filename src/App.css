.pAbsolute {
	position: absolute;
}

.pRelative {
	position: relative;
}

.dFlex {
	display: flex;
}

.flexDirectionColumn {
	flex-direction: column;
}

.justifyContentCenter {
	justify-content: center;
}

.flexBasis100 {
	flex-basis: 100%;
}

.flexGrow1 {
	flex-grow: 1;
}

.flexWrap {
	flex-wrap: wrap;
}

.r0 {
	right: 0;
}

.alignItemsCenter {
	align-items: center;
}

.alignItemsFlexEnd {
	align-items: flex-end;
}

.textAlignCenter {
	text-align: center;
}

.rotate90 {
	transform: rotate(90);
}

.w50 {
	width: 50%;
}

.w100 {
	width: 100%;
}

.elementMarginTop {
	margin-top: 30px;
}

.marginAllSites {
	margin: 30px;
}

.noMargin {
	margin: 0;
}

.headerBorderDetail {
	border-bottom: 10px solid;
}

.capitalize {
	text-transform: capitalize;
	font-weight: bold;
}

.fontSize24 {
	font-size: 24px;
}

.contentBox {
	padding: 10px 20px;
	border: 1px solid;
}
