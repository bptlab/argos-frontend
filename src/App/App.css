@font-face {
  font-family: "Ionicons";
  src: url("../assets/fonts/icons/ionicons.eot?v=2.0.0");
  src: url("../assets/fonts/icons/ionicons.eot?v=2.0.0#iefix") format("embedded-opentype"), url("../assets/fonts/icons/ionicons.ttf?v=2.0.0") format("truetype"), url("../assets/fonts/icons/ionicons.woff?v=2.0.0") format("woff"), url("../assets/fonts/icons/ionicons.svg?v=2.0.0#Ionicons") format("svg");
  font-weight: normal;
  font-style: normal; }

html, body, p, div, h1, h2, h3, h4, h5 {
  margin: 0;
  padding: 0; }
