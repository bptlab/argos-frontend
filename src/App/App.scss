$COLOR-success: #7CB342;
$COLOR-success-light: #689F38;
$COLOR-warning: #FFB300;
$COLOR-warning-light: #FFA000;
$COLOR-error: #F4511E;
$COLOR-error-light: #E64A19;

$ICON-errorQuantity: '\f3e5';
$ICON-serialNumber: '\f44c';
$ICON-date: '\f3f3';
$ICON-quantity: '\f44f';

@font-face { font-family: "Ionicons";
  src: url("../assets/fonts/icons/ionicons.eot?v=2.0.0");
  src: url("../assets/fonts/icons/ionicons.eot?v=2.0.0#iefix")
  format("embedded-opentype"), url("../assets/fonts/icons/ionicons.ttf?v=2.0.0")
  format("truetype"), url("../assets/fonts/icons/ionicons.woff?v=2.0.0")
  format("woff"), url("../assets/fonts/icons/ionicons.svg?v=2.0.0#Ionicons")
  format("svg"); font-weight: normal; font-style: normal; }


html, body, p, div, h1, h2, h3, h4, h5 {
  margin: 0;
  padding: 0;
}