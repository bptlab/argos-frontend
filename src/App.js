import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Header from './Header';
import './App.css';

class App extends Component {
    render() {
        return (
            <MuiThemeProvider>
                <Header />
            </MuiThemeProvider>
        );
    }
}

export default App;
