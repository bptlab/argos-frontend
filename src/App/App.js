import React, {Component} from 'react';
import ProductCardGrid from '../ProductCardGrid/ProductCardGrid.js';
import SearchBar from '../SearchBar/SearchBar.js'
import './App.css';
import './bootstrap.scss'
import Diagram from '../Diagram/Diagram'

class App extends Component {

    constructor(props) {
        super(props);
        this.state = {
            searchText: ''
        };
        this.handleUserInput = this.handleUserInput.bind(this)
    }

    handleUserInput(searchText) {
        this.setState({searchText: searchText});
    }

    componentDidMount() {
    }

    render() {
        let listOfPorducts = JSON.parse('[{"numberOfDevices":254,"numberOfEvents":2001,"productionStart":"Feb 1, 2016 12:00:00 AM","state":"RUNNING","name":"example family","id":47, "metaData": {"label":"product label 002", "brand":"Testbrand1", "orderNumber":1234, "statusDescription":"everything is broken!"}}, {"numberOfDevices":1337,"numberOfEvents":9001,"productionStart":"Feb 1, 4099 12:00:00 AM","state":"ERROR","name":"example family","id":42, "metaData": {"label":"product label 001", "brand":"Testbrand1", "orderNumber":1234, "statusDescription":"everything is broken!"}}]');
        return (
            <div className="AppWrapper">
                <Diagram products={listOfPorducts} />
                <SearchBar searchText={this.state.searchText} onUserSearchInput={this.handleUserInput} />
                <ProductCardGrid searchText={this.state.searchText} products={listOfPorducts} />
            </div>);
    }
}

export default App;
