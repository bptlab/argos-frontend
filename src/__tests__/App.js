import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App/App.js';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});
