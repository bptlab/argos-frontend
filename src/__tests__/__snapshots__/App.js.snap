exports[`test renders without crashing 1`] = `
<div>
  
</div>
`;

exports[`test renders without crashing 2`] = `
<div>
  <div
    className="alert-sliding alert-sliding-danger"
    role="alert">
    <table>
      <tbody>
        <tr>
          <td>
            <i
              className="fa fa-info-circle" />
          </td>
          <td>
            <p>
              Test-Notification
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
